### ISSUE TITLE
Binary Ordering Of Values

### ISSUE THREAD

**Issue by shauncksm** [shauncksm/base64id-rs#12]
## Problem
At this time any base64id value created has enforced upon it a binary sorting order, by way of [custom](https://docs.rs/base64id/latest/base64id/derive.Base64Id.html#partialord-ord) `PartialOrd` and `Ord` traits.

For unsigned integer values there is no special behavior as their sorting order is identical to binary sorting.

For signed integer values however, the integer is converted to it's unsigned counterpart and sorted based on that.

## Considerations
It's my opinion that a binary sorting order is the correct sorting order for the kind of value that base64id values represent; a container for a series of bytes with either decimal or base64url representation.

Given this opinion may prove problematic for users of this crate which may need to use alternate specific `PartialOrd` / `Ord` trait implementations, I see four options for how to resolve this.

## Solutions

1. **Do Nothing**. Simply leave the implementation as is. When thinking about the consideration above though, this isn't my favorite option. If a user did need to bring a custom `Ord` impl along, they simply can't due to conflicting trait impls.
2. **Remove The Binary Ord Trait**. This would resolve the issue and leave the user free to bring in whatever sorting order they needed. However, I still hesitate on this, as I do believe binary sorting is best.
3. **Make The Binary Ord Trait Opt In**. I believe this could be done like this:
```rust
#[derive(base64id::Base64Id)]
#[base64id(BinarySortOrd)
struct MyCustomId(i64);
```
The only real down side to this is that it would suddenly require all projects to decide upon a sorting order, picking the straightforward `#[derive(PartialOrd, Ord)]` that everybody knows or reapplying the binary sorting, which may require a bit of digging to figure out and apply (if the user wasn't expecting it).

4. **Make The Binary Ord Trait Opt Out**. This could be done like this:
```rust
#[derive(base64id::Base64Id)]
#[base64id(NoOrdTraits)
struct MyCustomId(i64);
```
I think this could be the most pragmatic option. _In my opinion it would make me happy_. More importantly though it'd mean users that actually have a need for specific ordering would then be able to apply their specific order. Mean while all other users would use the _correct_ / _default_ binary ordering.

## Feedback

Thoughts and feedback on this are welcome, as once a decision is made and a v1.0.0 is declared, it likely won't be possible to change this behavior again.

**Comment by shauncksm** [comment]
## `PartialOrd`, `Ord` Traits

In the end I was most favorable towards options 2 and 3 and decided to go with option 2, removing the custom order traits completely.

While this will require a migration effort for anyone using the ordering traits, I've come to believe its the most pragmatic option.
- This remove complexity from the library
- This frees up the user to add their own order traits as and if needed
- If the user really needed to binary sort a signed integer, I'd expect they'd do that on their project regardless of what this crate provided
- Ordering a value whose purpose is to serve as an opaque id, especially in the 64/128 bit sizes seems a bit pointless. If a binary (or some other ordering style) was highly expected I'd have been tempted to go with options 3 or 4, but given the kind of values base64id is meant to represent, I'm no longer convinced binary sorting (or any sorting for that matter) is appropriate

In summary this change simplifies this crate, and leaves the user free to pick whatever sorting method they need (if needed).

 ## `MIN` / `MAX` Constants

I've also decided to remove the MIN/MAX constants from the derive macro.
This is mainly due to the fact that, the very idea of a minimum and a maximum value depends upon some ordering style been in place. With ordering now optionally been supplied by the user, it's not possible for this crate to declare (via a constant) what the min and max values are.
